suite-native/navigation/src/components/NavigationContainerWithAnalytics.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactNode, useRef } from 'react';
+
+import { NavigationContainer, useNavigationContainerRef } from '@react-navigation/native';
+
+import { analytics, EventType } from '@suite-native/analytics';
+
+export const NavigationContainerWithAnalytics = ({ children }: { children: ReactNode }) => {
+    const navigationContainerRef = useNavigationContainerRef();
+    const routeNameRef = useRef<string | undefined>();
+
+    return (
+        <NavigationContainer
+            ref={navigationContainerRef}
+            onReady={() => {
+                routeNameRef.current = navigationContainerRef.getCurrentRoute()?.name;
+            }}
+            onStateChange={() => {
+                const previousRouteName = routeNameRef.current;
+                const currentRouteName = navigationContainerRef.getCurrentRoute()?.name;
+
+                if (previousRouteName !== currentRouteName) {
+                    // Save the current route name for later comparison
+                    routeNameRef.current = currentRouteName;
+
+                    if (!currentRouteName || !previousRouteName) return;
+
+                    analytics.report({
+                        type: EventType.ScreenChange,
+                        payload: {
+                            previousScreen: previousRouteName,
+                            currentScreen: currentRouteName,
+                        },
+                    });
+                }
```

Comment:
nit: please move this to handler outside JSX

Reply:
fixed in 7c3f39874